hw_diag/utilities/network_watchdog.py

```diff
@@ -0,0 +1,81 @@
+import time
+import urllib.request
+from datetime import datetime, timedelta
+from hm_pyhelper.logger import get_logger
+from retry.api import retry_call
+from utilities.balena_supervisor import BalenaSupervisor
+
+logging = get_logger(__name__)
+
+RESTART_FILE_NAME = 'restarts.csv'
+RESTART_DATE_FORMAT = '%d/%m/%Y %H:%M:%S'
+
+
+def have_internet() -> bool:
+    """wait for wait_time seconds for modem to achieve connectivity"""
+    try:
+        retry_call(
+            urllib.request.urlopen,
+            fkwargs={"url": "https://google.com", "timeout": 3},
+            tries=3,
+            delay=5,
+            logger=logging,
+        )
+        return True
+    except Exception as e:
+        logging.info(f"Internet is not accessible: {e}")
+        return False
+
+
+def get_last_restart() -> datetime:
+    try:
+        f = open(RESTART_FILE_NAME)
+        lines = f.read().splitlines()
+        last_line = lines[-1]
+        return datetime.strptime(last_line, RESTART_DATE_FORMAT)
+    except Exception as e:
+        return datetime.min
+    finally:
+        try:
+            f.close()
+        except:
+            pass
+
+
+def save_last_restart() -> None:
```

Comment:
This file will grow indefinitely. We need some kind of `logrotate` style implementation here.

Reply:
Make sense. For saving the last restart time and restart counters, I am not going to append them to the file but just write to the file in the volume. For the diagnostics purpose, I would like to configure the logging for the watchdog using `logrotate` in the persistent volume.